src/components/app/service.ts

```diff
@@ -187,17 +187,28 @@ const gitTriggersModal = (triggers, materials) => {
     })
 }
 
-const processMaterialHistory = (material) => {
+const processMaterialHistoryAndSelectionError = (material) => {
+    const data = {
+        isMaterialSelectionError: true,
+        materialSelectionErrorMsg: NO_COMMIT_SELECTED,
+        history: [],
+    }
     if (material.history) {
-        return material.history.map((history, index) => {
-            return {
+        let selectedIndex = -1
+        const matHistory = []
+        for (let index = 0; index < material.history.length; index++) {
+            const history = material.history[index]
+            if (selectedIndex === -1 && !history.Excluded) {
+                selectedIndex = index
+            }
+            matHistory.push({
```

Comment:
we can directly push to `data.history` instead of creating new `matHistory` array

Reply:
done